### DISCUSSION TITLE
Edit Customer Quote

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/Dynamics365Commerce.Solutions#254] [Q&A] [answered]
As per the requirement we are displaying the quotes in custom view, but while trying to edit these quotes we are facing issues.

We are currently using EditCustomerOrderOperationRequest in POS to edit customer orders, but same request is not supported in case of quotes. i.e., edited quotes using EditCustomerOrderOperationRequest is throwing errors. Also, we didn’t find any request in POS(which is exposed for customizations) to edit the quote.

We tried to use the approach of edit quotes from SearchOrdersView(Standard view), where in standard code system is editing the quotes using OperationManger class. This OperationManager class will eventually call the RecallQuoteCommerceServiceRequest to get the quote and assigns the cart response to Commerce.Session.instance.cart. But using customization, we can get the quotes(using custom RecallQuote API) but assigning the response to Commerce.Session.instance.cart is not possible(as Session.instance.cart is not exposed for customizations). Due to this, quotes are not getting added to cart screen and cart screen will be empty when navigated to CartView.

We also tried using CreateEmptyCartServiceRequest and BeginTransactionTrigger to assign the cart fetched through customized recall quote API, but cart screen was still empty as Commerce.Session.instance.cart was not set.

**Comment by ghost** [comment]
Need help with this implementation

**Comment by madyke** [comment, marked as answer]
Hello @ManishAcharya3986 - I looked into this, but unfortunately there isn't an existing extensibility point for editing a quote. And even after discussing it with the team, we weren't able to come up with any existing workarounds.

If this is a required business scenario, you'll need to submit an extensibility request so that the product team can review and triage the requirements and make a determination on what a possible timeline to implement it would be:
[Extensibility requests - Finance & Operations | Dynamics 365 | Microsoft Learn](https://learn.microsoft.com/en-us/dynamics365/fin-ops-core/dev-itpro/extensibility/extensibility-requests)

**Comment by ghost** [comment]
Hi @madyke I have already created an extensibility request #906481 for this. Created on 15-Feb-24.
